server/src/main/java/org/apache/gravitino/server/web/auth/jcasbin/GravitinoAdapter.java

```diff
@@ -0,0 +1,148 @@
+ * under the License.
+ */
+package org.apache.gravitino.server.web.auth.jcasbin;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import java.util.List;
+import org.apache.gravitino.storage.relational.po.CatalogPO;
+import org.apache.gravitino.storage.relational.po.MetalakePO;
+import org.apache.gravitino.storage.relational.po.SecurableObjectPO;
+import org.apache.gravitino.storage.relational.po.UserPO;
+import org.apache.gravitino.storage.relational.po.UserRoleRelPO;
+import org.apache.gravitino.storage.relational.service.CatalogMetaService;
+import org.apache.gravitino.storage.relational.service.MetalakeMetaService;
+import org.apache.gravitino.storage.relational.service.SecurableObjectService;
+import org.apache.gravitino.storage.relational.service.UserMetaService;
+import org.apache.gravitino.storage.relational.service.UserRoleRelService;
+import org.casbin.jcasbin.model.Model;
+import org.casbin.jcasbin.persist.Adapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The GravitinoAdapter reads the privilege policy from the database and adds it to the model of
+ * JCasbin.
+ */
+public class GravitinoAdapter implements Adapter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoAdapter.class);
+
+  @Override
+  public void loadPolicy(Model model) {
+    loadSecurableObject(model); // p
+    loadRoleUserRel(model); // g
+    loadGroupUserRel(model); // g
+    loadGroupRoleRel(model); // g
+  }
+
+  private void loadGroupRoleRel(Model model) {
+    LOG.info("load group role rel {}", model);
+    // TODO
+  }
+
+  private void loadGroupUserRel(Model model) {
+    LOG.info("load group user rel {}", model);
+    // TODO
+  }
+
+  private void loadRoleUserRel(Model model) {
+    List<UserRoleRelPO> userRoleRelList = UserRoleRelService.getInstance().listAllUserRoleRel();
+    for (UserRoleRelPO userRoleRelPO : userRoleRelList) {
+      UserPO userPO = UserMetaService.getInstance().getUserById(userRoleRelPO.getUserId());
+      Long metalakeId = userPO.getMetalakeId();
+      MetalakePO metalakePO = MetalakeMetaService.getInstance().getMetalakePOById(metalakeId);
+      model.addPolicy(
+          "g",
+          "g",
+          Lists.newArrayList(
+              String.valueOf(userRoleRelPO.getUserId()),
+              String.valueOf(userRoleRelPO.getRoleId()),
+              metalakePO.getMetalakeName()));
+    }
+  }
+
+  private void loadSecurableObject(Model model) {
+    List<SecurableObjectPO> securableObjectPOList = SecurableObjectService.getInstance().listAll();
+    securableObjectPOList.forEach(
+        securableObjectPO -> {
+          switch (securableObjectPO.getType()) {
+            case "CATALOG":
+              loadCatalog(securableObjectPO, model);
+              break;
+            default:
+              break;
+          }
+        });
+  }
```

Comment:
hi @hdygxsj Can we use this interface to implement a dynamic load of a single metadata permission?
for example:
```
loadSecurableObject(table_metadata, ...)
```

If we can do it, then we can support async load policy in to Jcasbin memory object.